lib/gym/package_command_generator.rb

```diff
@@ -45,6 +56,48 @@ def ipa_path
       def dsym_path
         Dir[BuildCommandGenerator.archive_path + "/**/*.dsym"].last
       end
+
+      private
+
+      # Fix PackageApplication Perl script by Xcode to create the IPA from the archive
+      def patch_package_application
+        require 'fileutils'
+
+        # Initialization
+        developer_dir = `xcode-select --print-path`.strip
+        patched_package_application_path = File.join("/tmp", "PackageApplication4Gym")
+        # Remove any previous patched PackageApplication
+        FileUtils.rm patched_package_application_path
+
+        Dir.mktmpdir do |tmpdir|
+          # Check current PackageApplication MD5
+          require 'digest'
+
+          expected_md5 = File.read("lib/assets/package_application_patches/PackageApplication_MD5")
+
+          raise "Found an invalid `PackageApplication` script. This is not supported." unless expected_md5 == Digest::MD5.file("#{developer_dir}/Platforms/iPhoneOS.platform/Developer/usr/bin/PackageApplication").hexdigest
+
+          # Duplicate PackageApplication script to PackageApplication4Gym
+          FileUtils.copy_file("#{developer_dir}/Platforms/iPhoneOS.platform/Developer/usr/bin/PackageApplication", patched_package_application_path)
```

Comment:
The path to the PackageApplication file can be found using

``````
xcrun --sdk iphoneos -f PackageApplication
``````

See https://gist.github.com/lacostej/f35f5fb8f3367551f22c#file-patch_packagapplication-sh-L12